### PR TITLE
Update libraries and fix crashes

### PR DESCRIPTION
**Pull request by mzorz**
Note: this PR sits on top of #1731 

### Fix
This app was not being consistently maintained so I had to bring a lot of libraries up to date and do some changes to make it all work as expected (also fixes a couple crashes, one of them listed [here](https://github.com/Automattic/simplenote-android/pull/1731#issuecomment-3137536096) from the previous PR)

### Test
Smoke test the app, create entires, verify they sync on another platform, check against the production app.

### Review
@ParaskP7 

### Release
Maintenance update - support for Android 15

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: PasscodeLock/src/main/java/org/wordpress/passcodelock/AbstractPasscodeKeyboardActivity.java [hunk 1]
```diff
@@ -24,9 +25,11 @@ public abstract class AbstractPasscodeKeyboardActivity extends Activity {
     protected InputFilter[] filters = null;
     protected TextView topMessage = null;
 
+	@SuppressLint("RestrictedApi")
```
Comment (mzorz) [inline PasscodeLock/src/main/java/org/wordpress/passcodelock/AbstractPasscodeKeyboardActivity.java:28]: Added this to avoid linter complaining for now, but tracking separately here https://github.com/Automattic/simplenote-android/issues/1734

File: PasscodeLock/build.gradle [hunk 2]
```diff
@@ -26,16 +26,21 @@ android {
             shrinkResources false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
+        releaseTest {
+            initWith release
+            minifyEnabled false
+            shrinkResources false
```
Comment (mzorz) [inline PasscodeLock/build.gradle]: had to create a different task to run tests on Release with minify disabled, since otherwise there was a ClassDefNotFound error on the AbstractAppLock class, and R8 was stripping the PasscodeLock classes.  Kudos to Claude Code for helping me out on this one, see https://github.com/Automattic/simplenote-android/pull/1733/commits/8c792b453529ad8997a80dada224f30c3da999c9

File: Simplenote/src/main/java/com/automattic/simplenote/NoteEditorActivity.java [hunk 3]
```diff
@@ -234,7 +234,8 @@ public void onPageScrollStateChanged(int state) {
 
     @Override
     public void onBackPressed() {
-        handleBackPressed();
+        super.onBackPressed();
```
[the remark this answers is not in the document]
Comment (mzorz) [reply Simplenote/src/main/java/com/automattic/simplenote/NoteEditorActivity.java:237]: this suggestion collides with the linter rule to call super in this deprecated method - I'm looking to make the minimum changes, a revamp to use getOnBackPressedDispatcher() consistently can be revisited later.

File: Simplenote/src/main/java/com/automattic/simplenote/authentication/NewCredentialsActivity.kt [hunk 4]
```diff
@@ -90,6 +90,7 @@ open class NewCredentialsActivity : AppCompatActivity() {
     }
 
     override fun onBackPressed() {
+        super.onBackPressed()
```
[the remark this answers is not in the document]
Comment (mzorz) [reply Simplenote/src/main/java/com/automattic/simplenote/authentication/NewCredentialsActivity.kt:93]: as said in https://github.com/Automattic/simplenote-android/pull/1733#discussion_r2250755187 I'm looking to do the minimum changes at this moment - calling super satisfies the linter error and the code after will be executed (it finishes the activity and launches a new one).

File: Simplenote/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -85,7 +86,7 @@
 
                 <data
                     android:host="app.simplenote.com"
-                    android:pathPattern="/account/.*/reset?redirect=simplenote://launch"
+                    android:pathPattern="/account/.*/reset"
```
[the remark this answers is not in the document]
Comment (mzorz) [reply Simplenote/src/main/AndroidManifest.xml:89]: there is additional validation in DeepLinkActivity - and again, looking to use the least changes at this moment.

File: gradlew [hunk 6]
```diff
@@ -114,7 +114,7 @@ case "$( uname )" in                #(
   NONSTOP* )        nonstop=true ;;
 esac
 
-CLASSPATH=$APP_HOME/gradle/wrapper/gradle-wrapper.jar
+CLASSPATH="\\\"\\\""
```
[the remark this answers is not in the document]
Comment (mzorz) [reply gradlew:117]: leaving this one up to @ParaskP7 🙏

File: gradlew.bat [hunk 7]
```diff
@@ -56,32 +59,34 @@ set JAVA_EXE=%JAVA_HOME%/bin/java.exe
...
+echo Please set the JAVA_HOME variable in your environment to match the 1>&2
+echo location of your Java installation. 1>&2
 
 goto fail
 
 :execute
 @rem Setup the command line
 
-set CLASSPATH=%APP_HOME%\gradle\wrapper\gradle-wrapper.jar
+set CLASSPATH=
```
[the remark this answers is not in the document]
Comment (mzorz) [reply gradlew.bat:75]: leaving this one up to @ParaskP7 🙏

File: gradlew (hunk 6 above)
[the remark this answers is not in the document]
Comment (ParaskP7) [reply gradlew:117]: Nah, this is not a manual change and we trust Gradle when updating these files automatically, can be ignored, but thanks bot! 😊

File: gradlew.bat (hunk 7 above)
[the remark this answers is not in the document]
Comment (ParaskP7) [reply gradlew.bat:75]: Nah, this is not a manual change and we trust Gradle when updating these files automatically, can be ignored, but thanks bot! 😊

**Comment by ParaskP7** [comment]
👋 @mzorz and just an FYI on my part that I started with reviewing this change today, but I didn't manage to finish with it. I'll continue tomorrow. ⏳

PS: In the meanwhile I am not sure if my review should be enough here, lots of changes and lots of library updates. On the other hand, it might be okay if we move faster (than usual) and only do a final check right at the end, when we're done with all the changes. What we want is to at least know that next time we'll need to release an update, we should be set. Wdyt? 🤔

**Comment by mzorz** [comment]
> PS: In the meanwhile I am not sure if my review should be enough here, lots of changes and lots of library updates. On the other hand, it might be okay if we move faster (than usual) and only do a final check right at the end, when we're done with all the changes. What we want is to at least know that next time we'll need to release an update, we should be set. Wdyt? 🤔

I'm fine either way, the main reason I split the PRs and asked for your review in this one is because this PR updates libraries and fixes crashes on the base PR (the one you wrote). The actual functional changes are on the other PRS (targeting SDK 35 plus changing the cwac:anddown library which is not 16kb-page-size compatible).

If you don't spot anything weird on this PR, this one should be good to go (since the PR we merged actually crashes the app anyway, this would be already putting things in a better place). 😅 

I'll have someone else review and smoke test the rest of the changes too 👍

**Comment by ParaskP7** [comment]
👋 @mzorz and I am on it! 🏃 💨 💨 💨

> I'm fine either way, the main reason I split the PRs and asked for your review in this one is because this PR updates libraries and fixes crashes on the base PR (the one you wrote). The actual functional changes are on the other PRS (targeting SDK 35 plus changing the cwac:anddown library which is not 16kb-page-size compatible).

Got it! 👍 

> If you don't spot anything weird on this PR, this one should be good to go (since the PR we merged actually crashes the app anyway, this would be already putting things in a better place). 😅

True... 😅 

> I'll have someone else review and smoke test the rest of the changes too 👍

Awesome, thanks! 🙇

File: .buildkite/pipeline.yml [hunk 8]
```diff
@@ -16,7 +16,7 @@ steps:
   - wait
 
   - label: ":hammer_and_pick: Build and Test"
-    command: ./gradlew --stacktrace testRelease
+    command: ./gradlew --stacktrace testReleaseTest
```
Comment (ParaskP7) [inline .buildkite/pipeline.yml]: Warning (⚠️): I am not sure why this `testRelease` task was chosen for CI builds (part of this initial 9d79c7296ce2e3c555045da4f8d5c29d209bb11b commit), and any `release` task for that matter, but I would suggest that instead us adding this new `releaseTest` build type, and all those new ProGuard configurations, that we instead change that to make CI run debug builds:

For the `Build and Test` specific job, to change to to:
```yml
  - label: ":hammer_and_pick: Unit Test"
    command: ./gradlew --stacktrace testDebugUnitTest
    plugins: [$CI_TOOLKIT_PLUGIN]
    notify:
      - github_commit_status:
          context: Unit Test
```

For the `Lint` specific job, to change that to:
```yml
  - label: ":microscope: Lint"
    command: ./gradlew --stacktrace lintDebug
    plugins: [$CI_TOOLKIT_PLUGIN]
    notify:
      - github_commit_status:
          context: Lint
```

FYI: The `Prototype Build`, which is using [build_and_upload_prototype_build](https://github.com/Automattic/simplenote-android/blob/trunk/fastlane/Fastfile#L136-L179), already builds on the [debug](https://github.com/Automattic/simplenote-android/blob/trunk/fastlane/Fastfile#L139) build type, not the `release`, so with this change everything will match `debug` for such development related builds.

PS: [release-build.yml](https://github.com/Automattic/simplenote-android/blob/trunk/.buildkite/release-build.yml) is dedicate to building releases, which point to build_for_distribution this [build_for_distribution](https://github.com/Automattic/simplenote-android/blob/trunk/fastlane/lanes/build.rb#L4-L31) that targets the [release](https://github.com/Automattic/simplenote-android/blob/trunk/fastlane/lanes/build.rb#L12C19-L12C26) build type and as such handles this case.

File: gradle/wrapper/gradle-wrapper.properties [hunk 9]
```diff
@@ -1,5 +1,7 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.14.3-bin.zip
```
Comment (ParaskP7) [inline gradle/wrapper/gradle-wrapper.properties]: 1. Suggestion (💡): Consider using the `all` distribution, which includes sources and docs, just like we do (almost) everywhere else, for consistency purposes.
2. Suggestion (💡): Consider adding the `distributionSha256Sum` for that specific distribution (see [release-checksums](https://gradle.org/release-checksums/)).

FYI: For your convenience feel free to use this command: `./gradlew wrapper --gradle-version=8.14.3 --distribution-type=all --gradle-distribution-sha256-sum=ed1a8d686605fd7c23bdf62c7fc7add1c5b23b2bbc3721e661934ef4a4911d7c && ./gradlew wrapper --distribution-type=all`

File: PasscodeLock/build.gradle [hunk 10]
```diff
@@ -26,16 +26,21 @@ android {
...
+        }
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
```
Comment (ParaskP7) [inline PasscodeLock/build.gradle:32]: Suggestion (💡): Just because I saw this, I remembered about the [.java-version](https://github.com/Automattic/simplenote-android/blob/trunk/.java-version) file, which it too is still pointing to Java `17.0`, and maybe we can update that to `21.0`, just like we did on DOAndroid quite recently.

File: PasscodeLock/build.gradle [hunk 11]
```diff
@@ -12,7 +12,7 @@ plugins {
 
 android {
```
Comment (ParaskP7) [inline PasscodeLock/build.gradle:13]: Suggestion (💡): Unrelated to the changes in this PR, but since we are trying to update it all, maybe we should consider updating `jcenter()` with `mavenCentral()` everywhere, see warning:

`JCenter Maven repository is no longer receiving updates: newer library versions may be available elsewhere`

File: PasscodeLock/build.gradle [hunk 12]
```diff
@@ -12,7 +12,7 @@ plugins {
...
     namespace 'org.wordpress.passcodelock'
-    compileSdk 34
+    compileSdk 36
 
     defaultConfig {
         minSdk 23
```
Comment (ParaskP7) [inline PasscodeLock/build.gradle:18]: Suggestion (💡): Again, somehow unrelated, but since we are here, let's remove this unnecessary `testInstrumentationRunner` from this module, it not having any `androidTest` and all.

File: PasscodeLock/build.gradle [hunk 13]
```diff
@@ -26,16 +26,21 @@ android {
...
 }
 
 dependencies {
-    implementation 'androidx.appcompat:appcompat:1.0.2'
-    implementation 'androidx.preference:preference:1.0.0'
+    implementation 'androidx.appcompat:appcompat:1.7.1'
+    implementation 'androidx.preference:preference:1.2.1'
 }
 
 android.buildTypes.all { buildType ->
```
Comment (ParaskP7) [inline PasscodeLock/build.gradle:41]: Suggestion (💡): Again, somehow unrelated, but since we are cleaning-up things, let's remove these unnecessary (or duplicate) [gradle.properties](https://github.com/Automattic/simplenote-android/blob/trunk/PasscodeLock/gradle.properties) configuration from this module, in the end, it seems that it could look like this:

`gradle.properties`
```properties
passcodelock.password_preference_key=passcode_lock_prefs_password_key
passcodelock.password_enc_secret=5-maggio-2002-Karel-Poborsky
passcodelock.fingerprint_enabled_key=passcode_lock_prefs_fingerprint_enabled_key

```

File: PasscodeLock/build.gradle [hunk 14]
```diff
@@ -26,16 +26,21 @@ android {
...
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
     }
 }
 
 dependencies {
-    implementation 'androidx.appcompat:appcompat:1.0.2'
-    implementation 'androidx.preference:preference:1.0.0'
+    implementation 'androidx.appcompat:appcompat:1.7.1'
```
Comment (ParaskP7) [inline PasscodeLock/build.gradle:37]: Suggestion (💡): Just like done [here](https://github.com/Automattic/simplenote-android/blob/trunk/build.gradle#L67-L74), I would recommend that now we are updating all those versions, we also extract all them to make sure we avoid updating one version and not the other, while at the same time make the first step towards using version catalogs (`libs.versions.toml`).

FYI: I can then take it from there and quickly, hopefully painlessly (famous last words), migrate this project to version catalogs. 

PS: Same for all those `min/compile/targetSdkVersion` values everywhere.

File: Simplenote/src/main/java/com/automattic/simplenote/widgets/SimplenoteEditText.java [hunk 15]
```diff
@@ -73,6 +73,13 @@ public boolean enoughToFilter() {
             return false;
         }
 
+		// solves a crash after updating dependencies in which this method
+	    // gets called in super() instantiation before the mTokenizer variable
+	    // is instantiated
+		if (mTokenizer == null) {
+			return false;
+		}
```
Comment (ParaskP7) [inline Simplenote/src/main/java/com/automattic/simplenote/widgets/SimplenoteEditText.java:81]: Minor (🔍): AS complains that the indentation is not correct here, because it uses tabs instead of empty spaces.

File: Simplenote/src/main/java/com/automattic/simplenote/DeepLinkActivity.kt [hunk 16]
```diff
@@ -27,8 +27,13 @@ class DeepLinkActivity : AppCompatActivity() {
                 startActivity(intent)
             }
             VERIFIED_WEB_SCHEME -> {
-                // New MagicLink
-                startMagicLinkConfirmation(uri)
+                // Check if this is a password reset URL
+                if (uri.path?.contains("/account/") == true && uri.path?.contains("/reset") == true) {
```
Comment (ParaskP7) [inline Simplenote/src/main/java/com/automattic/simplenote/DeepLinkActivity.kt:31]: ℹ️ Info: Acknowledging commit 28c7b258c8a6f27417808f6cf276f52c95591228 and change! 👍

File: Simplenote/src/main/java/com/automattic/simplenote/DeepLinkActivity.kt [hunk 17]
```diff
@@ -88,6 +93,21 @@ class DeepLinkActivity : AppCompatActivity() {
         }
     }
 
+    private fun handlePasswordReset(uri: Uri) {
+        val redirectParam = uri.getQueryParameter("redirect")
+        if (redirectParam == "simplenote://launch") {
```
Comment (ParaskP7) [inline Simplenote/src/main/java/com/automattic/simplenote/DeepLinkActivity.kt]: Question (❓): The logic within both `if/else` seems the same to me, what that done on purpose? 🤔

File: Simplenote/src/main/java/com/automattic/simplenote/NoteEditorActivity.java [hunk 18]
```diff
@@ -249,7 +250,7 @@ private void handleBackPressed() {
 
             finish();
         } else {
-            super.onBackPressed();
+	        getOnBackPressedDispatcher().onBackPressed();
```
Comment (ParaskP7) [inline Simplenote/src/main/java/com/automattic/simplenote/NoteEditorActivity.java:253]: ℹ️ Info: Acknowledging commit 22fefdef25267033af1b4d2b07a1fd2d1ecaaad5 and change! 👍

File: Simplenote/src/main/res/layout/empty_view.xml [hunk 19]
```diff
@@ -17,7 +18,7 @@
         android:layout_marginTop="@dimen/padding_large"
         android:layout_width="@dimen/icon_empty"
         android:src="@drawable/ic_notes_24dp"
-        android:tint="?attr/emptyImageColor">
+        app:tint="?attr/emptyImageColor">
     </ImageView>
```
Comment (ParaskP7) [inline Simplenote/src/main/res/layout/empty_view.xml:22]: Minor (🔍): Just an FYI that there are multiple other places where such an empty `</ImageView>` tag can be collapsed into `/>` (3 in here, another 3 within `fragment_note_editor.xml`, 5 within `note_list_row.xml`, etc).

File: Simplenote/src/test/java/com/automattic/simplenote/viewmodels/CollaboratorsViewModelTest.kt [hunk 20]
```diff
@@ -144,16 +144,26 @@ class CollaboratorsViewModelTest {
 
     @Test
     fun collaboratorAddedAfterStoppedListeningChangesShouldNotUpdateUiState() = runTest {
```
Comment (ParaskP7) [inline Simplenote/src/test/java/com/automattic/simplenote/viewmodels/CollaboratorsViewModelTest.kt:146]: 1. Praise (❤️): Thanks for fixing this unit test! 🥇 
2. Warning (⚠️): In addition to having unit test working, let's consider having instrumented tests working as well, in this or another follow-up PR (if not too much of a trouble). On that side of things I have the following:
    1. First, we need to add this `androidTestImplementation 'androidx.test.uiautomator:uiautomator:2.3.0'` dependency to have those instrumented tests compile.
    2. Then, we need to use `./gradlew Simplenote:connectedDebugAndroidTest` over `./gradlew connectedDebugAndroidTest` to make them run, because otherwise compilation fails on the `PasscodeLock` module with multiple duplicate class failures on Kotlin (we might consider getting to the bottom of this).
    3. Finally, after managing to run instrumented tests successfully, you'll find that at least `13/74` fail.

**Review by ParaskP7 — CHANGES_REQUESTED** [review]
👋 @mzorz !

I have reviewed and tested this PR as per the instructions, great job updating all these dependencies! 🌟 x 🌟 ^ 🌟

-----

I have left a couple of warnings (⚠️), one question (❓), a few suggestions (💡) and some minor (🔍) comments for you to consider. I am **NOT** going to approve this PR just yet, mainly because of this one CI related warning (⚠️), but other than that I think this is good to go, we can keep iterating on modernizing this codebase on subsequent PRs.

File: Simplenote/src/main/java/com/automattic/simplenote/DeepLinkActivity.kt (hunk 17 above)
Comment (mzorz) [reply Simplenote/src/main/java/com/automattic/simplenote/DeepLinkActivity.kt]: oof good catch, fixed in 2589709f - it should just launch the app when capturing a password reset link and the app is installed (this to do the same thing it was doing before, no behavior change)

Comment (ParaskP7) [reply Simplenote/src/main/java/com/automattic/simplenote/DeepLinkActivity.kt]: Awesome, thanks for the fix @mzorz , so in the end, no query parameters no nothing needed here, right? 😊

File: .buildkite/pipeline.yml (hunk 8 above)
Comment (mzorz) [reply .buildkite/pipeline.yml]: Good question - the `testRelease` task was running tests on Release, but R8 is stripping off classes from PasscodeLock in the generated build being tested, so the suggestion I took from Claude was to create a new flavor that didn't have minify enabled so the classes would exist and tests would pass, that's why . See https://github.com/Automattic/simplenote-android/commit/8c792b453529ad8997a80dada224f30c3da999c9 -  I can see that we could continue to play with the proguard arguments and perhaps set minify to true in release just to use the same task that existed before, but I wonder if we can. Thoughts?

**Comment by ParaskP7** [comment]
FYI: I was able to reproduce [this exact problem](https://github.com/Automattic/simplenote-android/pull/1737#issuecomment-3154067905) while testing this PR, but only did it once, then I just couldn't any more... 🤷

File: PasscodeLock/build.gradle (hunk 11 above)
Comment (mzorz) [reply PasscodeLock/build.gradle:13]: 👍 done in 9229a7a6

Comment (ParaskP7) [reply PasscodeLock/build.gradle:13]: Awesome, thanks @mzorz ! 🙇 ❤️ 🚀

File: PasscodeLock/build.gradle (hunk 10 above)
Comment (mzorz) [reply PasscodeLock/build.gradle:32]: done in 6f51438f

File: PasscodeLock/build.gradle (hunk 12 above)
Comment (mzorz) [reply PasscodeLock/build.gradle:18]: done in e9447a50 - any efforts regarding instrumented tests I'd say let's tackle separately

File: PasscodeLock/build.gradle (hunk 14 above)
Comment (mzorz) [reply PasscodeLock/build.gradle:37]: sounds like a great idea, can we do that on a different PR?

File: .buildkite/pipeline.yml (hunk 8 above)
Comment (ParaskP7) [reply .buildkite/pipeline.yml]: 1. Yea, no, I understood all that @mzorz , Claude is playing smart here... 😅 But, I honestly don't think we need to run unit tests on release, thus just changing this `testRelease` to `testDebug`, then making it explicit, changing `testDebug` to `testDebugUnitTest` might be the best we can do here.
2. Also, I am in favor of the above, because I would try to avoid anything ProGuard related in this PR, and just for testing purposes, let alone creating a new build type, just for that. I really think we should avoid these changes unless totally necessary.

File: PasscodeLock/build.gradle (hunk 10 above)
Comment (ParaskP7) [reply PasscodeLock/build.gradle:32]: Awesome, thanks @mzorz ! 🙇 ❤️ 🚀

File: Simplenote/src/main/java/com/automattic/simplenote/widgets/SimplenoteEditText.java (hunk 15 above)
Comment (mzorz) [reply Simplenote/src/main/java/com/automattic/simplenote/widgets/SimplenoteEditText.java:81]: good call fixed in 7a4fb340

File: PasscodeLock/build.gradle (hunk 12 above)
Comment (ParaskP7) [reply PasscodeLock/build.gradle:18]: Awesome, thanks @mzorz ! 🙇 ❤️ 🚀

File: PasscodeLock/build.gradle (hunk 14 above)
Comment (ParaskP7) [reply PasscodeLock/build.gradle:37]: Totally @mzorz , resolving this! 💯

File: Simplenote/src/main/res/layout/empty_view.xml (hunk 19 above)
Comment (mzorz) [reply Simplenote/src/main/res/layout/empty_view.xml:22]: that's right - I only fixed those where I also visited to fix something else. We can fix them all separately.

File: Simplenote/src/main/java/com/automattic/simplenote/widgets/SimplenoteEditText.java (hunk 15 above)
Comment (ParaskP7) [reply Simplenote/src/main/java/com/automattic/simplenote/widgets/SimplenoteEditText.java:81]: Great, thanks @mzorz ! 🙇 ❤️ 🚀

File: Simplenote/src/main/res/layout/empty_view.xml (hunk 19 above)
Comment (ParaskP7) [reply Simplenote/src/main/res/layout/empty_view.xml:22]: 💯 this is truly minor (🔍) @mzorz and I only mention it because you already tried to do something similar, resolving this now... 😊

File: gradle/wrapper/gradle-wrapper.properties (hunk 9 above)
Comment (mzorz) [reply gradle/wrapper/gradle-wrapper.properties]: thanks much for the pointers here! done in daae06db

Comment (ParaskP7) [reply gradle/wrapper/gradle-wrapper.properties]: Awesome, thank you so much @mzorz ! 🙇 ❤️ 🚀

**Comment by mzorz** [comment]
Thanks a lot for all this feedback @ParaskP7 ! 🙌  really appreciated.

> FYI: I was able to reproduce https://github.com/Automattic/simplenote-android/pull/1737#issuecomment-3154067905 while testing this PR, but only did it once, then I just couldn't any more... 🤷

Thanks for flagging - all of that is being addressed either on the PR changing to `targetSdk 35` or later on but on top of that PR. 

Just so we're on the same page: for this PR, I concentrated on:
- upgrading the libraries to latest (not the target SDK)
- making sure what exists doesn't crash

This PR is already large enough with that aim (upgrading the libraries versions and updating code to match behavior changes only).

So, I fixed some of the comments that were related to the changes above, but all further suggestions I think are better to be handled in separate PRs.

> Warning (⚠️): In addition to having unit test working, let's consider having instrumented tests working as well, in this or another follow-up PR (if not too much of a trouble).

Thanks for the pointers here! Also for instrumented tests, I agree it'd be nice to bring them back to life, yet it exceeds the aim of updating the libraries only. I also can imagine integrated tests will change after the changes we're making due to the edge-to-edge UI changes by targeting SDK 35, so let's keep that task isolated for the time being.

I think the only really blocking thing in this PR should be [this](https://github.com/Automattic/simplenote-android/pull/1733#discussion_r2253878342). Let's continue the discussion there 👍

**Comment by ParaskP7** [comment]
💯 @mzorz and I agree with all that you said [above](https://github.com/Automattic/simplenote-android/pull/1733#issuecomment-3155054999), thank you so much for applying all and any of my suggestions! 🙇

FYI: I've resolved all comments but did left the ones we chose to differ as unresolved, just to make it more visible going forward, when and if we decide to do something about them. 🤞

> I think the only really blocking thing in this PR should be https://github.com/Automattic/simplenote-android/pull/1733#discussion_r2253878342. Let's continue the discussion there 👍

Yes, let's do that and then merge this! 🚀

File: .buildkite/pipeline.yml (hunk 8 above)
Comment (mzorz) [reply .buildkite/pipeline.yml]: Got it - it took me 5 extra minutes to grasp these concepts better and think I understand your suggestion now 😅. I removed the unneeded flavors and updated the yaml to get buildkite to run unit tests on debug in this commit here 3616f1ac and 0448e630.

Comment (ParaskP7) [reply .buildkite/pipeline.yml]: Awesome @mzorz , as soon as CI goes 🟢 this is ready to merge, thank you so much! 🙇 ❤️ 🚀

**Review by ParaskP7 — APPROVED** [review]

**Comment by mzorz** [comment]
Thanks once more for you thorough review @ParaskP7 ! We'll take over the things that were left unresolved in subsequent PRs as agreed 🙌

File: PasscodeLock/build.gradle (hunk 13 above)
Comment (mzorz) [reply PasscodeLock/build.gradle:41]: Following up on this - tried to remove the file but these are used in BuildConfig which is referenced in DefaultAppLock. The keys are actually used as Android prefs keys, so we need to keep them otherwise changing them it means an app upgrade would be reading a different setting. So, keeping them for now 👍

File: Simplenote/src/test/java/com/automattic/simplenote/viewmodels/CollaboratorsViewModelTest.kt (hunk 20 above)
Comment (mzorz) [reply Simplenote/src/test/java/com/automattic/simplenote/viewmodels/CollaboratorsViewModelTest.kt:146]: Moved this one out to a ticket here https://github.com/Automattic/simplenote-android/issues/1744

File: PasscodeLock/build.gradle (hunk 13 above)
Comment (ParaskP7) [reply PasscodeLock/build.gradle:41]: Thanks for checking @mzorz but just to make it clear, I suggested to remove everything else but these 3 from this [gradle.properties](https://github.com/Automattic/simplenote-android/blob/trunk/PasscodeLock/gradle.properties) file, is that what you tried? 😊

`before`
```
android.enableJetifier=true
android.useAndroidX=true

passcodelock.password_preference_key=passcode_lock_prefs_password_key
passcodelock.password_enc_secret=5-maggio-2002-Karel-Poborsky
passcodelock.fingerprint_enabled_key=passcode_lock_prefs_fingerprint_enabled_key

ossrhUsername=hello
ossrhPassword=world

signing.keyId=byebye
signing.password=secret
signing.secretKeyRingFile=/home/user/secret.gpg
```

`after`
```
passcodelock.password_preference_key=passcode_lock_prefs_password_key
passcodelock.password_enc_secret=5-maggio-2002-Karel-Poborsky
passcodelock.fingerprint_enabled_key=passcode_lock_prefs_fingerprint_enabled_key
```

File: Simplenote/src/test/java/com/automattic/simplenote/viewmodels/CollaboratorsViewModelTest.kt (hunk 20 above)
Comment (ParaskP7) [reply Simplenote/src/test/java/com/automattic/simplenote/viewmodels/CollaboratorsViewModelTest.kt:146]: Awesome, thanks @mzorz , marking this as resolved now. 🙇 ❤️

File: PasscodeLock/build.gradle (hunk 13 above)
Comment (mzorz) [reply PasscodeLock/build.gradle:41]: OH I misread that. You're right we can keep those ones only. PR https://github.com/Automattic/simplenote-android/pull/1745

Comment (ParaskP7) [reply PasscodeLock/build.gradle:41]: Thanks so much @mzorz , #1745 comment resolved! 🙇 ✅ ❤️ 🚀
